docs/workforce-management-framework/topics/con-workflow-step.adoc

```diff
@@ -0,0 +1,60 @@
+= Workflow Step
+
+== Step Structure
+Steps are divided into to linked catergories the `module` and the `definition`.
+
+=== Module
+Modules can contain one or more angular directives for rendering steps in the portal and mobile applications. 
+{WFM-RC-NameShort} steps use Angular.js directives to build user interface features as part of workflows.
+Each step is divided into two types of directive:
+
+- **Edit Section (Form)**
+
+The Edit section is displayed when workflow is executed and allows users to edit information typically this would be a form.
+The Edit section consists of the html template and directive.
+
+- **Preview Section (View)**
+
+The Preview section is displayed as a summary when a workflow has been completed. The Preview section
+consists of the html template and directive.
+
+=== Definition 
+Definition contains the step definition which outlines the properties such as name and decription of a step for the workflow in the portal application.
+Each step requires a definition so the `definition.json` file can contain a single definition or 
+an array of definitions depending on the number of steps that are included in the step package.
+
+Example:
+[source,json]
+----
+[
+    {
+        "code": "base-form",
+        "name": "Base Form",
+        "description": "base form used to for steps",
+        "templates": {
+            "form": "<base-form></base-form>",
+            "view": "<base></base>"
+        }
+    }
+]
+----
+The module and the definition are exported to be visibile when step package is imported. 
```

Comment:
Finding it hard to understand the full content in ^ line.....

This is what I'm thinking:

- The module and the definition are set to be visible when exported as a _Step_ package which means that when they are imported into another _Workflow_, they are visible to the user.

Reply:
@jstaffor - this content is basically impossible to understand without basic knowledge of angular and our application structure. 
@austincunningham - as mentioned before please make sure that we mention that at the beginning.

Reply:
this is how elements are exposed in node see https://www.sitepoint.com/understanding-module-exports-exports-node-js/ for context.